src/android/Dgram.java

```diff
@@ -75,198 +48,191 @@ public void run() {
+        try {
+            JSONObject payload = new JSONObject();
+
+            payload.put("error", e.toString());
+
+            CallbackUtil.sendPluginResult(onMessageCallback,
+                    new PluginResult(PluginResult.Status.ERROR, payload)
+            );
+        } catch (Exception exception) {
+            Log.e(TAG, "Exception generating payload:" + exception.toString());
+            PluginResult result = new PluginResult(
+                    PluginResult.Status.ERROR,
+                    "Error occurred while creating onMessage payload: " + e.toString()
+            );
+            CallbackUtil.sendPluginResult(onMessageCallback, result);
+        }
+    }
+
     private class DatagramSocketSend implements Runnable {
-        String m_message;
-        String m_address;
-        int m_port;
-        int m_datagramSocketId;
-        DatagramSocket m_datagramSocket;
-
-        public DatagramSocketSend(
-                final String message,
-                final String address,
-                final int port,
-                final int datagramSocketId,
-                final DatagramSocket datagramSocket
-        ) {
-            this.m_message = message;
-            this.m_address = address;
-            this.m_port = port;
-            this.m_datagramSocketId = datagramSocketId;
-            this.m_datagramSocket = datagramSocket;
+        CallbackContext sendCallback;
+        String message;
+        String address;
+        int port;
+
+        public DatagramSocketSend(CallbackContext callbackContext, final String message,
+                final String address, final int port) {
+            this.sendCallback = callbackContext;
+            this.message = message;
+            this.address = address;
+            this.port = port;
         }
 
         public void run() {
             try {
-                if (this.m_datagramSocket.isClosed()) {
+                if (datagramSocket.isClosed()) {
                     Log.d(TAG, "Trying to send but socket closed");
                     return;
                 }
 
                 // Threaded send to prevent NetworkOnMainThreadException
-                final byte[] bytes = this.m_message.getBytes("UTF-8");
+                final byte[] bytes = this.message.getBytes(StandardCharsets.UTF_8);
                 final DatagramPacket packet = new DatagramPacket(
                         bytes,
                         bytes.length,
-                        InetAddress.getByName(this.m_address),
-                        this.m_port
+                        InetAddress.getByName(this.address),
+                        this.port
                 );
-                this.m_datagramSocket.send(packet);
-                Dgram.this.m_javascriptQueue.put("_onSend(" + this.m_datagramSocketId + ");");
-            } catch (InterruptedException e) {
-                Log.d(TAG, e.toString());
+                datagramSocket.send(packet);
```

Comment:
i find using the `datagramSocket here as a global member from the enclosing class confusing.